managed-ledger/src/main/java/org/apache/bookkeeper/mledger/impl/PositionImpl.java

```diff
@@ -102,20 +99,27 @@ public PositionImpl getNext() {
      */
     @Override
     public String toString() {
-        return String.format("%d:%d", ledgerId, entryId);
+        return ledgerId + ":" + entryId;
     }
 
     @Override
-    public int compareTo(PositionImpl other) {
-        checkNotNull(other);
+    public int compareTo(PositionImpl that) {
+        if (this.ledgerId != that.ledgerId) {
+            return (this.ledgerId < that.ledgerId ? -1 : 1);
```

Comment:
`java.lang.Long#compare(this.ledgerId, other.ledgerId)` is more readable.

Reply:
It's not equivalent, I replied in https://github.com/apache/pulsar/pull/14679#discussion_r825958570